src/main/java/frc/robot/subsystems/Intake.java

```diff
@@ -56,9 +70,11 @@ public void periodic() {
         boolean noteAtShooterSensor = ! shooterSensor.get();
         switch (currentState) {
             case IDLE:
-                if (noteAtPreIntakeSensor) {
+                if (noteAtPreIntakeSensor || noteAtIntakeSensor) {
                     //intakeMotor.setControl(velocity.withVelocity(intakeMotorVelocity));
-                    intakeMotor.set(0.2);
+                    if (! isEjecting) {
+                        intakeMotor.set(0.2);
+                    }
```

Comment:
This is the other side of the comment about trying to go backwards with your state machine. You had to use this code AND have the `||` above change because  of it. Now you have two paths through your state machine, one forward, one backward. That's totally valid for a state machine but it really can get confusing the way this one is designed. I recommend adding new enum values that are like `EJECTING_AT_INTAKE` or `EJECTING_AT_HOLDING` and use that instead of this `isEjecting` flag as well.